artifactory.py

```diff
@@ -1015,14 +1036,50 @@ def del_properties(self, pathobj, props, recursive):
     def scandir(self, pathobj):
         return _ScandirIter((pathobj.joinpath(x) for x in self.listdir(pathobj)))
 
-    def writeto(self, fd, out, chunk_size):
-        url = str(fd)
-        res = fd.session.get(url, stream=True, verify=True, cert=None)
-        if res.status_code != 200:
-            raise RuntimeError(res.status_code)
-        for chunk in res.iter_content(chunk_size=chunk_size):
-            if chunk:
-                out.write(chunk)
+    def writeto(
+        self,
+        arti_cls,
```

Comment:
What have you renamed `fd` to `arti_cls`? If it's class, why can't we use `ArtifactoryPath` below?

Reply:
We run the function below like `self._accessor.writeto(self, output, chunk_size, progress_func, *args, **kwargs)`

I guess, it must be `fd`, not `arti_cls`

Reply:
Oh, got it, I've forgot that it is not `ArtifactoryPath`

But `arti_cls` doesn't fit here. It's more like `artifactory_path_instance`, but let's keep it simple and use `fd`?

Reply:
OR `pathobj` like in `get_response`

Reply:
> OR `pathobj`

this one I like, will redo